src/SriServiceProvider.php

```diff
@@ -29,39 +30,57 @@ public function boot()
         ]);
 
         Blade::directive('mixSri', function (string $path, bool $crossOrigin = false) {
+            $path = $this->removeQuotes($path);
+
             if (Str::startsWith($path, ['http', 'https', '//'])) {
                 $href = $path;
             } else {
                 $href = mix($path);
             }
 
-            $integrity = SriFacade::html($path, $crossOrigin);
-
-            if (Str::endsWith($path, 'css')) {
-                return "<link href='{$href}' rel='stylesheet' {$integrity}>";
-            } elseif (Str::endsWith($path, 'js')) {
-                return "<script src='{$href}' {$integrity}></script>";
-            } else {
-                throw new \Exception('Invalid file');
-            }
+            return $this->parseAndGenerateUrl($path, $href, $crossOrigin);
         });
 
         Blade::directive('assetSri', function (string $path, bool $crossOrigin = false) {
+            $path = $this->removeQuotes($path);
+
             if (Str::startsWith($path, ['http', 'https', '//'])) {
                 $href = $path;
             } else {
                 $href = asset($path);
             }
 
-            $integrity = SriFacade::html($path, $crossOrigin);
-
-            if (Str::endsWith($path, 'css')) {
-                return "<link href='{$href}' rel='stylesheet' {$integrity}>";
-            } elseif (Str::endsWith($path, 'js')) {
-                return "<script src='{$href}' {$integrity}></script>";
-            } else {
-                throw new \Exception('Invalid file');
-            }
+            return $this->parseAndGenerateUrl($path, $href, $crossOrigin);
         });
     }
+
+    private function removeQuotes(string $path): string
+    {
+        $values = ['\'', '"'];
+
+        return str_replace($values, '', $path);
+    }
+
+    private function parseAndGenerateUrl(string $path, string $href, bool $crossOrigin)
```

Comment:
Missing the return statement here ;) 

```suggestion
    private function parseAndGenerateUrl(string $path, string $href, bool $crossOrigin): HtmlString
```